kdoctor/src/commonMain/kotlin/org/jetbrains/kotlin/doctor/Main.kt

```diff
@@ -3,45 +3,55 @@ package org.jetbrains.kotlin.doctor
 import co.touchlab.kermit.CommonWriter
 import co.touchlab.kermit.Logger
 import co.touchlab.kermit.Severity
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.default
+import com.github.ajalt.clikt.parameters.arguments.optional
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
 import kotlinx.coroutines.runBlocking
 import org.jetbrains.kotlin.doctor.entity.System
 
 internal const val KDOCTOR_VERSION = "0.0.6"
 
-fun main(args: Array<String>) {
-    val arguments = Arguments(args)
-
-    Logger.setLogWriters(object : CommonWriter() {
-        override fun isLoggable(severity: Severity) = arguments.isDebug
-    })
-
-    when {
-        arguments.isShowVersion -> {
-            println(KDOCTOR_VERSION)
-        }
-
-        else -> {
-            diagnoseKmmEnvironment(
-                arguments.isVerbose,
-                arguments.isFull,
-                arguments.projectPath,
-                arguments.localCompatibilityJson.takeIf { arguments.isDebug }
-            )
+fun main(args: Array<String>) = Main().main(args)
+
+private class Main : CliktCommand() {
+    val showVersion: Boolean by option(
+        "--version",
+        help = "Report a version of KDoctor"
+    ).flag()
+    val isVerbose: Boolean by option(
+        "--verbose", "-v",
+        help = "Report an extended information"
+    ).flag()
+    val isExtraDiagnostics: Boolean by option(
+        "--all", "-a",
+        help = "Run extra diagnostics such as a build of a synthetic project and an analysis of a project in the current directory"
+    ).flag()
+
+    val isDebug: Boolean by option("--debug", hidden = true).flag()
+    val localCompatibilityJson: String? by option("--compatibilityJson", hidden = true)
+    val templateProjectTag: String? by option("--templateProject", hidden = true)
+
+    override fun run() {
+        Logger.setLogWriters(object : CommonWriter() {
+            override fun isLoggable(severity: Severity) = isDebug
+        })
+
+        when {
+            showVersion -> {
+                println(KDOCTOR_VERSION)
+            }
+
+            else -> runBlocking {
+                Doctor(getSystem())
+                    .diagnoseKmmEnvironment(isVerbose, isExtraDiagnostics, localCompatibilityJson, templateProjectTag)
```

Comment:
why? all arguments are named already.

Reply:
Because it's easy to mix up the order of parameters while refactoring/editing this code. First three parameters here have the same type so compiler/ide won't help with this.

Also when writing code myself I always add names to call-site when function has more than 2-3 params. Just feel more safe that way.